docs/writing_plugins.md

```diff
@@ -166,22 +168,47 @@ for (const file of schema.files) {
 
 ### Printing to a generated file
  
-As mentioned, the object returned from `generateFile` contains a `print` function which is a variadic function accepting zero-to-many string values.  These values will then be "printed" to the file so that when the actual physical file is generated by the compiler, all values given to `print` will be included in the file.  Successive strings passed in the same invocation will be appended to one another.  To print an empty line, pass zero arguments to `print`.
+As mentioned, the object returned from `generateFile` contains a `print` function which can be used to print your generated code to a file.  The `print` function is an overloaded function which can be used in one of two ways:
+
+#### As a variadic function
+
+The first way is as a variadic function which accepts zero-to-many string arguments.  These values will then be "printed" to the file so that when the actual physical file is generated by the compiler, all values given to `print` will be included in the file.  Successive strings passed in the same invocation will be appended to one another.  To print an empty line, pass zero arguments to `print`.
  
 For example:
 
 ```ts
 const name = "UserService";
 f.print("export class ", name, "Client {");     
-f.print();
+f.print("    console.log('Hello world');");
 f.print("}");
 ```
 
 The above will generate:
 
 ```ts
 export class UserServiceClient {
+    console.log('Hello world');
+}
+```
+
+#### As a template literal tagged function
+
+You can also pass a template literal to the function and use string interpolation as you would do in regular JavaScript:
+
+For example:
 
+```ts
+const name = "UserService";
+f.print`export class ${name}Client {     
+    console.log('Hello world');
+}`;
```

Comment:
Ah, hold on. Should we really recommend multiline strings? This code will always be in a function body: 

```ts
  const name = "UserService";
  f.print`export class ${name}Client {     
      console.log('Hello world');
  }`;
```

Which means it will generate:

```ts
export class UserServiceClient {
      console.log('Hello world');
  }
```

(extra leading whitespace on line 2 and 3)

Reply:
Gotcha.  Updated.